File: Serene/Serene.Script/Membership/ResetPasswordPanel.cs

namespace Serene.Membership
{
    using jQueryApi;
    using Serenity;
    using System.Html;

    [Panel, FormKey("Membership.ResetPassword")]
    public class ResetPasswordPanel : PropertyPanel<ResetPasswordRequest>
    {
        private ResetPasswordForm form;

        public ResetPasswordPanel(jQueryObject container)
            : base(container)
        {
            form = new ResetPasswordForm(this.IdPrefix);

            form.NewPassword.AddValidationRule(this.UniqueName, e =>
            {
                if (form.ConfirmPassword.Value.Length < 7)
                    return string.Format(Q.Text("Validation.MinRequiredPasswordLength"), 7);

                return null;
            });

            form.ConfirmPassword.AddValidationRule(this.UniqueName, e =>
            {
                if (form.ConfirmPassword.Value != form.NewPassword.Value)
                    return Q.Text("Validation.PasswordConfirm");

                return null;
            });

            this.ById("SubmitButton").Click((s, e) =>
            {
                e.PreventDefault();

                if (!ValidateForm())
                    return;

                var request = GetSaveEntity();
                request.Token = this.ById("Token").GetValue();
                Q.ServiceCall(new ServiceCallOptions
                {
                    Url = Q.ResolveUrl("~/Account/ResetPassword"),
                    Request = request,
                    OnSuccess = response =>
                    {
                        Q.Information(Q.Text("Forms.Membership.ResetPassword.Success"), () =>
                        {
                            Window.Location.Href = Q.ResolveUrl("~/Account/Login");
                        }, new ConfirmOptions());
                    }
                });
            });
        }
    }
}

File: Serene/Serene.Core/Modules/Membership/Account/ResetPassword/ResetPasswordPanel.ts
namespace Serene.Membership {

    @Serenity.Decorators.registerClass()
    export class ResetPasswordPanel extends Serenity.PropertyPanel<ResetPasswordRequest, any> {

        protected getFormKey() { return ResetPasswordForm.formKey; }

        private form: ResetPasswordForm;

        constructor(container: JQuery) {
            super(container);

            this.form = new ResetPasswordForm(this.idPrefix);

            this.form.NewPassword.addValidationRule(this.uniqueName, e => {
                if (this.form.NewPassword.value.length < 7) {
                    return Q.format(Q.text('Validation.MinRequiredPasswordLength'), 7);
                }
            });

            this.form.ConfirmPassword.addValidationRule(this.uniqueName, e => {
                if (this.form.ConfirmPassword.value !== this.form.NewPassword.value) {
                    return Q.text('Validation.PasswordConfirm');
                }
            });

            this.byId('SubmitButton').click(e => {
                e.preventDefault();

                if (!this.validateForm()) {
                    return;
                }

                var request = this.getSaveEntity();
                request.Token = this.byId('Token').val();
                Q.serviceCall({
                    url: Q.resolveUrl('~/Account/ResetPassword'),
                    request: request,
                    onSuccess: response => {
                        Q.information(Q.text('Forms.Membership.ResetPassword.Success'), () => {
                            window.location.href = Q.resolveUrl('~/Account/Login');
                        });
                    }
                });

            });
        }
    }
}